spark-common/src/main/java/me/lucko/spark/common/command/modules/SamplerModule.java

```diff
@@ -428,12 +428,17 @@ private void handleUpload(SparkPlatform platform, CommandResponseHandler resp, S
                 String url = platform.getViewerUrl() + key;
 
                 resp.broadcastPrefixed(text("Profiler stopped & upload complete!", GOLD));
-                resp.broadcast(text()
-                        .content(url)
-                        .color(GRAY)
-                        .clickEvent(ClickEvent.openUrl(url))
-                        .build()
-                );
+                if (resp.sender().hasPermission("spark.viewurl")) {
```

Comment:
Added an if block that checks the user's `spark.viewurl` permission value and either prints the URL or prints an info message depending on the value.